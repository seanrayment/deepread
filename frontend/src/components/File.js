import React, { Component } from "react";
var moment = require('moment')

class File extends Component{
    constructor(props){
        super(props);
    }
    handleClick = () => {
        this.props.selectFile(this.props.file.pk); 
    }

    render() {
        return(
            <tr onClick={this.handleClick}>
                <td>
                    <p>{this.props.file.title}</p>
                </td>
                <td>
                    <div className="file-metadata">
                        <p className="file-metadata-descriptor">last modified <span>{moment(this.props.file.updated_at).fromNow() }</span></p>
                        <p className="file-metadata-descriptor">created on <span>{moment(this.props.file.created_at).format('MMMM D YYYY')}</span></p>
                    </div>
                </td>
            </tr>
        );
    }
}

export default File;