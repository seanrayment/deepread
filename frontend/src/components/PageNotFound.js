import React, { Component } from "react";
import illustration from "../undraw_web_search_eetr.svg";
class PageNotFound extends Component {
    
    render() {
        return (
            <div className="notfound-body">
                <div className="notfound-content">
                    <h1>Page not found</h1>
                    <div className="notfound-card">
                        <p>Either you don't have access to the requested resource or it does not exist.</p>                        
                    </div>
                    <img className="notfound-illustration" src={illustration}></img>
                </div>
            </div>
        )
    }
}
export default PageNotFound;