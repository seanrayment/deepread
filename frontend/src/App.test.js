import  React   from 'react';
import ReactDOM from 'react-dom';
import Login from './components/Login';
import PageNotFound from './components/PageNotFound'
import SignUp from './components/SignUp'
import File from './components/File'
import renderer from 'react-test-renderer';

it('renders login without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Login />, div);
});

it('renders signup without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<SignUp />, div);
});

it('renders 404 without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<PageNotFound />, div);
});


test('x displays when hovering file', () => {
  let file = {
    title: 'My Test File',
    updated_at: 0,
    created_at: 0,
  }
  const component = renderer.create(
    <File file={file}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseOver();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
