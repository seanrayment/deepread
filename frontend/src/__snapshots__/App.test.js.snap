// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`x displays when hovering file 1`] = `
<tr
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <td>
    <p
      className="file-title"
      onClick={[Function]}
    >
      My Test File
    </p>
  </td>
  <td>
    <div
      className="file-metadata"
    >
      <p
        className="file-metadata-descriptor"
      >
        last modified 
        <span>
          50 years ago
        </span>
      </p>
      <p
        className="file-metadata-descriptor"
      >
        created on 
        <span>
          December 31 1969
        </span>
      </p>
    </div>
  </td>
  <td>
    <div
      className="file-dot-menu-wrap"
    />
  </td>
</tr>
`;

exports[`x displays when hovering file 2`] = `
<tr
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <td>
    <p
      className="file-title"
      onClick={[Function]}
    >
      My Test File
    </p>
  </td>
  <td>
    <div
      className="file-metadata"
    >
      <p
        className="file-metadata-descriptor"
      >
        last modified 
        <span>
          50 years ago
        </span>
      </p>
      <p
        className="file-metadata-descriptor"
      >
        created on 
        <span>
          December 31 1969
        </span>
      </p>
    </div>
  </td>
  <td>
    <div
      className="file-dot-menu-wrap"
    >
      <div
        className="file-dots"
      >
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "#212121",
              "height": "24px",
              "width": "24px",
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M3 9.5a1.5 1.5 0 110-3 1.5 1.5 0 010 3zm5 0a1.5 1.5 0 110-3 1.5 1.5 0 010 3zm5 0a1.5 1.5 0 110-3 1.5 1.5 0 010 3z"
            fillRule="evenodd"
          />
        </svg>
        <div
          className="file-menu"
          onClick={[Function]}
        >
          <p>
            Delete document
          </p>
        </div>
      </div>
    </div>
  </td>
</tr>
`;

exports[`x displays when hovering file 3`] = `
<tr
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <td>
    <p
      className="file-title"
      onClick={[Function]}
    >
      My Test File
    </p>
  </td>
  <td>
    <div
      className="file-metadata"
    >
      <p
        className="file-metadata-descriptor"
      >
        last modified 
        <span>
          50 years ago
        </span>
      </p>
      <p
        className="file-metadata-descriptor"
      >
        created on 
        <span>
          December 31 1969
        </span>
      </p>
    </div>
  </td>
  <td>
    <div
      className="file-dot-menu-wrap"
    />
  </td>
</tr>
`;
